codegen-test/model/misc.smithy

```diff
@@ -108,3 +112,67 @@ union AUnion {
     string: String,
     time: Timestamp,
 }
+
+/// This operation tests that the response code defaults to @http's code
+@http(method: "GET", uri: "/responseCodeDefault", code: 418)
+@httpRequestTests([
+    {
+        id: "ResponseCodeDefaultTest",
+        protocol: "aws.protocols#restJson1",
+        uri: "/responseCodeDefault",
+        method: "GET",
+    }
+])
+@httpResponseTests([
+    {
+        id: "ResponseCodeDefaultTest",
+        protocol: "aws.protocols#restJson1",
+        code: 418,
+    }
+])
+operation ResponseCodeDefault {
+    input: ResponseCodeDefaultInput,
+    output: ResponseCodeDefaultOutput,
+}
+
+@input
+structure ResponseCodeDefaultInput {}
+
+@output
+structure ResponseCodeDefaultOutput {}
+
+/// This operation tests that @httpResponseCode is @required
+/// and is used over @http's code
+@http(method: "GET", uri: "/responseCodeRequired", code: 200)
+@httpRequestTests([
```

Comment:
This should be an `httpResponseTests` (add it to the array below). There are no response codes in HTTP requests.